pkg/service/FluxService/helper.go

```diff
@@ -0,0 +1,363 @@
+	if err != nil || !found {
+		return nil, fmt.Errorf("status not found")
+	}
+
+	inventory, found, err := unstructured.NestedMap(status, INVENTORY)
+	if err != nil || !found {
+		return nil, fmt.Errorf("inventory not found")
+	}
+
+	entries, found, err := unstructured.NestedSlice(inventory, ENTRIES)
+	if err != nil || !found {
+		return nil, fmt.Errorf("entries not found")
+	}
+
+	for _, item := range entries {
+		itemMap, ok := item.(map[string]interface{})
+		if !ok {
+			return nil, fmt.Errorf("invalid item format")
+		}
+
+		metadataCompact := ObjectMetadataCompact{}
+		if id, found, err := unstructured.NestedString(itemMap, ID); found && err == nil {
+			metadataCompact.Id = id
+		} else if err != nil {
+			return nil, err
+		}
+
+		if version, found, err := unstructured.NestedString(itemMap, VERSION); found && err == nil {
+			metadataCompact.Version = version
+		} else if err != nil {
+			return nil, err
+		}
+		fluxManagedResourcesMap[metadataCompact.Id] = metadataCompact.Version
+	}
+	return fluxManagedResourcesMap, nil
+}
+func getReleaseNameNamespace(obj map[string]interface{}, name string) (string, string, error) {
+	//storageNamespace is optional, if not found, then its value byb default is the same as the namespace of helmRelease
+	var storageNamespace string
+
+	if statusRawObj, ok := obj[STATUS]; ok {
+		statusObj := statusRawObj.(map[string]interface{})
+		if storageNamespaceRaw, ok6 := statusObj["storageNamespace"]; ok6 {
+			storageNamespace = storageNamespaceRaw.(string)
+		} else {
+			metadata, found, err := unstructured.NestedMap(obj, "metadata")
+			if err != nil {
+				return "", "", err
+			}
+			if !found {
+				return "", "", fmt.Errorf("entries not found")
+			}
+			namespaceRaw, found, err := unstructured.NestedFieldNoCopy(metadata, "namespace")
+			if err != nil {
+				return "", "", err
+			}
+			if !found {
+				return "", "", fmt.Errorf("entries not found")
+			}
+			storageNamespace = namespaceRaw.(string)
+		}
+	}
+
+	//releaseName is also optional, but if not provided then releaseName is decided by combination of "<targetNamespace>-name" and targetNamespace is optional,in absence of optional targetNamespace, releaseName will be same as HelmRelease.
+
+	spec, found, err := unstructured.NestedMap(obj, "spec")
+	if err != nil {
+		return "", "", err
+	}
+	if !found {
+		return "", "", fmt.Errorf("entries not found")
+	}
+
+	releaseNameRaw, found, err := unstructured.NestedFieldCopy(spec, "releaseName")
+	if err != nil {
+		return "", "", err
+	}
+	if found {
+		return releaseNameRaw.(string), storageNamespace, nil
+	}
+
+	targetNamespaceRaw, found, err := unstructured.NestedFieldNoCopy(spec, "targetNamespace")
+	if err != nil {
+		return "", "", err
+	}
+	if found {
+		return targetNamespaceRaw.(string) + "-" + name, storageNamespace, nil
+	}
+
+	return name, storageNamespace, nil
+
+}
+func getKsAppStatus(obj map[string]interface{}, gvk schema.GroupVersionKind) (*FluxAppStatusDetail, error) {
```

Comment:
change func. name to getFluxAppStatus, since we are returning flux Helm release status also